packages/ui/src/components/TimeInputV2/index.tsx

```diff
@@ -0,0 +1,469 @@
+    height: ${({ theme }) => theme.sizing[INPUT_SIZE_HEIGHT.medium]};
+  }
+
+  &[data-size='large'] {
+    height: ${({ theme }) => theme.sizing[INPUT_SIZE_HEIGHT.large]};
+  }
+
+  &[data-readonly='true'] {
+    background: ${({ theme }) => theme.colors.neutral.backgroundWeak};
+    border-color: ${({ theme }) => theme.colors.neutral.border};
+    cursor: default;
+  }
+
+  &[data-disabled='true'] {
+    background: ${({ theme }) => theme.colors.neutral.backgroundDisabled};
+    border-color: ${({ theme }) => theme.colors.neutral.borderDisabled};
+    cursor: not-allowed;
+    user-select: none;
+  }
+
+  &[data-error='true'] {
+    border: 1px solid ${({ theme }) => theme.colors.danger.border};
+
+    &:not([data-disabled="true"]):not([data-readonly="true"]):active {
+      border-color: ${({ theme }) => theme.colors.danger.borderHover};
+      box-shadow: ${({ theme }) => theme.shadows.focusDanger};
+    }
+
+    &:not([data-disabled="true"]):not([data-readonly="true"]):hover {
+      border-color: ${({ theme }) => theme.colors.danger.borderHover};
+    }
+  }
+`
+
+export const Input = styled.input<{
+  'data-size': 'small' | 'medium' | 'large'
+  'data-readonly': boolean
+  'data-disabled': boolean
+  'data-period'?: boolean
+}>`
+  border: none;
+  outline: none;
+  background: transparent;
+  font-size: ${({ theme }) => theme.typography.bodySmall.fontSize};
+  width: ${({ theme }) => theme.sizing[312]};
+  height: ${({ theme }) => theme.sizing[300]};
+  text-align: center;
+  border-radius: ${({ theme }) => theme.radii.default};
+  color: ${({ theme }) => theme.colors.neutral.text};
+
+  &[data-size='large'] {
+    font-size: ${({ theme }) => theme.typography.body.fontSize};
+  }
+
+  &:not([data-disabled="true"]):not([data-readonly="true"]):hover {
+    background-color: ${({ theme }) => theme.colors.neutral.backgroundHover};
+    color: ${({ theme }) => theme.colors.neutral.textWeak};
+  }
+
+  &:not([data-disabled="true"]):not([data-readonly="true"]):active, 
+  :not([data-disabled="true"]):not([data-readonly="true"]):focus{
+    background-color: ${({ theme }) => theme.colors.neutral.backgroundStrong};
+    color:  ${({ theme }) => theme.colors.neutral.text};
+  }
+
+  &[data-readonly='true'] {
+    cursor: default;
+  }
+
+  &[data-disabled='true'] {
+    cursor: not-allowed;
+    user-select: none;
+  }
+
+  &[data-period="true"] {
+    color: ${({ theme }) => theme.colors.neutral.textWeak};
+  }
+
+  ::-moz-selection {
+    background: none;
+  }
+
+  ::selection {
+    background: none;
+  }
+`
+
+const CustomText = styled(Text)`
+padding-inline: ${({ theme }) => theme.space['0.25']};
+`
+
+type TimeInputProps = {
+  label?: ReactNode
+  placeholder?: Time
+  value?: Time
+  clearable?: boolean
+  required?: boolean
+  labelDescription?: ReactNode
+  helper?: string
+  disabled?: boolean
+  readOnly?: boolean
+  error?: boolean | string
+  'data-testid'?: string
+  onChange?: (time: Time, timePeriod?: string) => void
```

Comment:
```suggestion
  onChange?: (value: Time, valuePeriod?: string) => void
```

To be consistent between our inputs